examples/llama7b_sparse_quantized/README.md

```diff
@@ -0,0 +1,46 @@
+# Creating a Sparse Quantized Llama7b Model
+
+The example in this folder runs in multiple stages to create a Llama 7b model with 
+a 2:4 sparsity pattern and W4A16 quantization. The model is calibrated and trained with 
+the Ultra Chat 200k dataset.
+
+## Recipe Summary
+
+The recipe used for this flow is located in [2:4_w4a16_recipe.yaml](./2:4_w4a16_recipe.yaml). It contains 3 stages that are outlined below.
+
+
+### Stage 1: Sparsification
+
+Runs the SparseGPT one-shot algorithm to prune the model to 50% sparsity with a 2:4 
+sparsity pattern. This means that 2 weights out of every group of 4 weights are masked to 0.
+
+### Stage 2: Finetuning Recovery
+
+This stage runs a single epoch of training on the ultrachat200k dataset while maintaining 
+the sparsity mask from stage 1. The purpose of this stage is to recover any accuracy lost 
+during the sparsification process.
+
+### Stage 3: Quantization
+
+Finally, we run the GPTQ one-shot algorithm to quantize all linear weights to 4 bit 
+channelwise.
+
+## How to Run
+
+We can run the entire staged recipe with one call to SparseML's `compress` pathway. This 
+will save a checkpoint of the model after each stage.
+
+```python examples/llama7b_sparse_quantized/llama7b_sparse_w4a16.py```
+
+### Compression
+
+The resulting model will be uncompressed. To save a final compressed copy of the model 
```

Comment:
Is that right? So the `compress` pathway does not use `compressed-tensors` when saving the optimized model to `output_dir = "output_llama7b_2:4_w4a16_channel_compressed"`?

Reply:
So we only save in compressed format if `save_compressed=True` is passed to sparseml.train/oneshot/apply/compress. This is really a naming issue we should address: we decided to call the multi_stage pathway "compress()" before compressed-tensors existed. Its a completely different pathway from ModelCompressor.compress() which actually compresses the model on disk. sparseml.compress() just runs oneshot and or finetuning. I'll update it to use the apply() function instead which is equivalent and change the output_dir name to make this less confusing.

I chose for this example to save in an uncompressed format to avoid running compression on all the checkpoints